python/nano/test/pytorch/tests/test_channels_last.py

```diff
@@ -0,0 +1,225 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import pytest
+import torch
+import torchmetrics
+from unittest import TestCase
+from bigdl.nano.pytorch import Trainer
+from torchvision.models.resnet import ResNet, BasicBlock
+from bigdl.nano.pytorch.lightning import LightningModuleFromTorch
+from bigdl.nano.pytorch.utils import TORCH_VERSION_LESS_1_12
+from test.pytorch.utils._train_torch_lightning import create_data_loader, data_transform
+from test.pytorch.utils._train_torch_lightning import create_test_data_loader
+
+num_classes = 10
+batch_size = 32
+dataset_size = 256
+num_workers = 0
+data_dir = os.path.join(os.path.dirname(__file__), "../data")
+
+class customResNet(ResNet):
+    def __init__(self):
+        super(customResNet, self).__init__(BasicBlock, [2, 2, 2, 2])
+        self.head = torch.nn.Linear(self.fc.out_features, num_classes)
+
+    def forward(self, x):
+        x = self.conv1(x)
+        x = self.bn1(x)
+        x = self.relu(x)
+        x = self.maxpool(x)
+
+        x = self.layer1(x)
+        x = self.layer2(x)
+        x = self.layer3(x)
+        x = self.layer4(x)
+        if not TORCH_VERSION_LESS_1_12:
+            assert x.is_contiguous(memory_format=torch.channels_last)
+        
+        x = self.avgpool(x)
+        x = torch.flatten(x, 1)
+        x = self.fc(x)
+        x = self.head(x)
+
+        return x
+
+class ConvModel(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = torch.nn.Conv2d(2, 1, (1, 2), bias=False)
```

Comment:
For Convolutional layer with only one channel, the layer.param.is_contiguous() & layer.param.is_contiguous(memory_format=torch.channels_last) is True after the model is converted to channels_last.